internal/openssh/lib.go

```diff
@@ -0,0 +1,214 @@
+package openssh
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/binary"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// This module cherry-picks from https://go-review.googlesource.com/c/crypto/+/218620,
+// that provides a solution for https://github.com/golang/go/issues/37132, that highlights
+// the need for methods to marshal private keys to the OpenSSH format in `x/crypto/ssh`.
+//
+// This provides utilities to serialize a Private Keys in OpenSSH format: once
+// this goes mainstream, we can remove this module.
```

Comment:
Nit: Can we please create the followup GitHub issue for tracking and reference it here please?

Reply:
Good idea

Reply:
Created [this](https://github.com/hashicorp/terraform-provider-tls/issues/154): will add in the comment.